framework/include/utils/MooseTypes.h

```diff
@@ -104,7 +104,7 @@ class Point;
 /**
  * Convenience macros for automatic dual/non-dual typing
  */
-#define ADReal typename RealType<compute_stage>::type
+#define ADReal Moose::typename RealType<compute_stage>::type
```

Comment:
Wow, that syntax is rank! Good thing we're hiding it in a macro. What does it clash with?